release/ray_release/scripts/ray_bisect.py

```diff
@@ -217,5 +227,26 @@ def _get_commit_lists(passing_commit: str, failing_commit: str) -> List[str]:
     )
 
 
+def _update_test_state(test: Test, blamed_commit: str) -> None:
```

Comment:
any unit test covering this function?

Reply:
I moved the bulk of the logic to state machine class and tested there